### PR TITLE
Bump to 1.1.2 - fixes JSON parse error

### PR DESCRIPTION
**Pull request by cbetta**

### CODE REVIEW

File: src/HttpClient.js [hunk 1]
```diff
@@ -80,11 +88,15 @@ class HttpClient {
...
+                      return callback({
+                        message: "The API response could not be parsed.",
+                        parseError: parsererr
+                      }, null);
 	                }
                 }
 
                 if(response.statusCode < 200 || response.statusCode > 299) {
-                  err = retJson;
+                  callback(err, retJson);
```
Comment (leggetter) [inline src/HttpClient.js]: Could this result in two callbacks being made? After this `callback` is executed the code would run to the next `callback`.

**Review by leggetter — DISMISSED** [review]
In general the code is now a bit harder to follow IMO since it has 3 (or maybe even 4 if `return` is added before all `callback` invocations) points of return.

Can it be simplified?

File: src/HttpClient.js (hunk 1 above)
Comment (cbetta) [reply src/HttpClient.js]: Yes it would. I will fix.

**Comment by cbetta** [comment]
@leggetter yes it could be, by adding actual tests to this class. Currently there are none.

**Comment by cbetta** [comment]
Just kidding. Rewriting the whole thing and adding a few tests.

**Comment by cbetta** [comment]
@leggetter mind giving he new version some fresh eyes?

**Review by leggetter — APPROVED** [review]

**Pull request merged**